File: webmanager/handlerevents/createnode/createnode.js
// Обработчик события (создание узла (папки))
function createnode() {
	var ls = decodeURIComponent(g_fullpath).split(',');
    if (ls.length > 4) {
		alert('На этом уровне невозможно создать папку!');
	    return;
	}
	var pr = false;
    if (g_sysid == 1)  // ЛТМ
		if (ls.length >= 4) {foldername = prompt('Введите номер КП', 'КП№xx'); pr = true;}
	if (!pr) foldername = prompt('Введите название создаваемой папки', 'NewFolder');
	
	if (foldername == null) return;
	if (foldername.trim() == '') return;    // Если пробелы, выходим
    $.getJSON('http://' + g_rhost + '/createnode' + '?path=' + g_fullpath + '&foldername=' + encodeURIComponent(foldername),
        function(data) {
			if (data == 'notfound') {       // Директория не найдена, кто-то удалил эту папку пока браузер был открыт
			    alert('Директория, в которой создается папка не найдена. Попробуйте обновить дерево!');
				return;
			}
			if (data == 'exists') {         // Папка уже существует (может другой клиент создал папку)
			    alert('Папка с таким именем уже существует!');
				return;    
			}
			if (data == 'notcreated') {      // Ошибка при создании папки
			    alert('Не удалось создать папку. Возможно нет прав на создание папки!');
				return;    
			}
			if (data == 'success') {         // Папка успешно создана на сервере
				$('#treeview').jstree('refresh');  // Обновляем дерево
				return;
			}
			console.log(data);
			alert('Сервер вернул неизвестное значение. Операция не выполнена');
		}
    )
}

File: webmanager/handlerevents/createnode/createnode.py
import os
import urllib

# Класс обработчик событий (создание узла)
class CreateNode:
    # Конструктор   
    def __init__(self, app, web):
        app.add_routes([web.get('/createnode', self.createnode)])        # Добавление маршрута
    
    async def createnode(self, request):
        params = request.rel_url.query
        fullpath = self.get_fullpath(params['path'], request.remote)     # Полный путь к выделенной папке
        if not os.path.exists(fullpath):                                 # Нет выделенной папки
            return self.web.json_response('notfound')

        dir_ = fullpath + '/' + urllib.parse.unquote(params['foldername'])
        if os.path.exists(dir_):                                         # Папка с таким именем уже существует
            return self.web.json_response('exists') 
        os.mkdir(dir_)
        if not os.path.exists(dir_):                                     # Не удалось создать папку
            return self.web.json_response('notcreated')
        
        return self.web.json_response('success')                         # Сообщаем, что все ОК
